CFG post-order panic: reflect: Elem of invalid type

The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	str := strconv.Itoa(101)
	fmt.Println(str[0] == '0')
}
```

Expected result:
```console
$ go run ./sample.go
false
```

Got:
```console
$ yaegi ./sample.go
panic: reflect: Elem of invalid type [recovered]
	panic: do.go:10:14: CFG post-order panic: reflect: Elem of invalid type

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2.1(0xc0000efd00)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/cfg.go:356 +0xba
panic(0x1894e20, 0x1b8c480)
	runtime/panic.go:679 +0x1b2
reflect.(*rtype).Elem(0x1894e20, 0x1bb9960, 0x3)
	reflect/type.go:920 +0x143
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc0000efd00)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/cfg.go:623 +0x5864
github.com/containous/yaegi/interp.(*node).Walk(0xc0000efd00, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:120 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc0000efc00, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:117 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ef800, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:117 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ef700, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:117 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000eef00, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:117 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000eea00, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:117 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000ee300, 0xc000195ce8, 0xc000195d20)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:117 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc0000f0a00, 0xc0000ee300, 0xc0000cc9e0, 0x4, 0x0, 0x0, 0x0)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/cfg.go:40 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000f0a00, 0xc0000fa100, 0x72, 0x272, 0xc0000fa100, 0x72, 0x0, 0x1)
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/interp/interp.go:267 +0x1ed
main.main()
	github.com/containous/yaegi@v0.6.3-0.20191020025203-15686873e081/cmd/yaegi/yaegi.go:82 +0x42d
```

If I change `str := strconv.Itoa(101)` to `var str string = strconv.Itoa(101)`, yaegi will run it without problem.

Tested on several functions that returns a string, like `os.Getenv`, `runtime.Version`, `io.EOF.Error`, and yaegi panics on all cases.

--- a/_test/str3.go
+++ b/_test/str3.go
@@ -1,0 +1,11 @@
+package main
+
+import "strconv"
+
+func main() {
+	str := strconv.Itoa(101)
+	println(str[0] == '1')
+}
+
+// Output:
+// true

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -634,7 +634,11 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 			case stringT:
 				n.typ = sc.getType("byte")
 			case valueT:
-				n.typ = &itype{cat: valueT, rtype: t.rtype.Elem()}
+				if t.rtype.Kind() == reflect.String {
+					n.typ = sc.getType("byte")
+				} else {
+					n.typ = &itype{cat: valueT, rtype: t.rtype.Elem()}
+				}
 			default:
 				n.typ = t.val
 			}

